src/pydisagg/ihme/splitter.py

```diff
@@ -101,6 +92,234 @@ def remove_prefix(self) -> None:
+    data: DataConfig
+    pattern: SexPatConfig
+    population: SexPopulationConfig
+
+    def model_post_init(self, __context: Any) -> None:
+        """Extra validation of all the index."""
+        if not set(self.pattern.by).issubset(self.data.index):
+            raise ValueError("pattern.by must be a subset of data.index")
+        if not set(self.population.index).issubset(
+            self.data.index + self.pattern.index
+        ):
+            raise ValueError(
+                "population.index must be a subset of data.index + pattern.index"
+            )
+
+    def _merge_with_pattern(
+        self, data: DataFrame, pattern: DataFrame
+    ) -> DataFrame:
+        data_with_pattern = data.merge(
+            pattern, on=self.pattern.by, how="left"
+        ).dropna()
+        return data_with_pattern
+
+    def parse_data(self, data: DataFrame) -> DataFrame:
+        name = "data"
+        validate_columns(data, self.data.columns, name)
+        data = data[self.data.columns].copy()
+        validate_index(data, self.data.index, name)
+        validate_nonan(data, name)
+        validate_positive(data, [self.data.val_sd], name)
+        return data
+
+    def parse_pattern(self, data: DataFrame, pattern: DataFrame) -> DataFrame:
+        name = "pattern"
+
+        if not all(
+            col in pattern.columns
+            for col in [self.pattern.val, self.pattern.val_sd]
+        ):
+            if not self.pattern.draws:
+                raise ValueError(
+                    "Must provide draws for pattern if pattern.val and "
+                    "pattern.val_sd are not available."
+                )
+            validate_columns(pattern, self.pattern.draws, name)
+            pattern[self.pattern.val] = pattern[self.pattern.draws].mean(axis=1)
+            pattern[self.pattern.val_sd] = pattern[self.pattern.draws].std(
+                axis=1
+            )
+
+        validate_columns(pattern, self.pattern.columns, name)
+        pattern = pattern[self.pattern.columns].copy()
+        validate_index(pattern, self.pattern.index, name)
+        validate_nonan(pattern, name)
+        validate_positive(pattern, [self.pattern.val_sd], name)
+        # rename_map = self.pattern.apply_prefix()
+        # pattern.rename(columns=rename_map, inplace=True)
+        data_with_pattern = self._merge_with_pattern(data, pattern)
+        return data_with_pattern
+
+    # This is currently hardcoded put we should somehow match index to index
+    # Perhaps additional params for the population config?
+    def parse_population(
+        self, data: DataFrame, population: DataFrame
+    ) -> DataFrame:
+        name = "population"
+        validate_columns(population, self.population.columns, name)
+
+        # Filter for each sex_id and select only the 'val' column, ignoring sex_id == 3
+        male_population = population[population[self.population.sex] == 1][
+            self.population.index + [self.population.val]
```

Comment:
Please see the comment above, here we hard coded user will use `1` represent male and `2` represent female